### DISCUSSION TITLE
Adapting Type Hinting for PyPSA-Earth

### DISCUSSION THREAD

**Discussion by GbotemiB** [pypsa-meets-earth/pypsa-earth#937] [Ideas]
Type hinting is a way to indicate expected types of variables in function arguments. 
Type hinting doesn't affect the actual runtime of code implementations. However, they are very useful in providing developers with additional information about types of data.

Some other advantage of type hinting aside from readability is that it is a form of documentation on its own. It also makes collaboration easier.

Happy to hear everyone's thought on this.

**Comment by pz-max** [comment]
Sounds good! I would recommend to wait until the merge of the sector & electricity model happens type hinting will effect the whole code base through out. There are also other folks that already started adding type hinting which helps us with the implementation.

**Reply by GbotemiB** [reply]
Thanks @pz-max. I was considering whether to have it implemented as I am currently working on Monte-Carlo. I will continue my implementation with type-hinting then.

**Reply by pz-max** [reply]
Yeah for the Monte-Carlo you can proceed with Typehinting no issue

**Comment by davide-f** [comment]
Agree that it is not of a priority.
I've read that it type hinting is also an interesting feature desired for linopy, which suggests that it is worth considering for the ecosystem
